### PR TITLE
RSDK-7370 fully static RDK

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- add `make full-static` target with `-extldflags=-static`
- add full-static.yml which builds this for 3 CPUs (amd64 arm64 arm) + tests it on alpine (a non-libc platform)
    - this workflow is not triggered yet anywhere. see 'follow-up work' below
    - :heavy_check_mark: [example run](https://github.com/viamrobotics/rdk/actions/runs/8883537628/job/24390599038) with 3x artifacts attached
- clean up no_cgo tag so it is truly no cgo
## Why
Create a baseline of RDK that runs 'everywhere' (for some definition of everywhere)
## Follow-up work
- the GCS upload logic here is untested and not used yet. it will be easier to test after merging this piece (because manual workflow_dispatch doesn't work until yml is on main branch)
- web/server/entrypoint_* needs cleanup -- there are too many variations now

### CODE REVIEW

**Comment by dgottlieb** [comment]
> -extldflags=-static

If I understand correctly, this flag means the linker will choose to error if it cannot statically link some library?

And the rest of the PR was about getting the system to satisfy that flag?

**Comment by abe-winter** [comment]
yup exactly -- you'll get many pages of linker errors, example:
```
go build -tags osusergo,netgo -ldflags="-extldflags=-static -s -w -X 'go.viam.com/rdk/config.Version=' -X 'go.viam.com/rdk/config.GitRevision=3a7515f0ecc0fb69a31e4397cfb1f307c459ac4e' -X 'go.viam.com/rdk/config.DateCompiled=2024-04-29'" -o bin/static/viam-server-amd64 ./web/cmd/server
# go.viam.com/rdk/web/cmd/server
/home/linuxbrew/.linuxbrew/Cellar/go@1.21/1.21.8/libexec/pkg/tool/linux_amd64/link: running cc failed: exit status 1
/usr/bin/ld: /tmp/go-link-1919731928/000040.o: in function `ma_dlopen':
miniaudio.c:(.text+0x4947): warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /home/linuxbrew/.linuxbrew/lib/libnlopt.a(stogo.cc.o): in function `stogo_minimize':
stogo.cc:(.text+0x1e3): undefined reference to `operator delete[](void*)'
/usr/bin/ld: stogo.cc:(.text+0x1eb): undefined reference to `operator delete(void*)'
/usr/bin/ld: stogo.cc:(.text+0x201): undefined reference to `operator delete[](void*)'
/usr/bin/ld: stogo.cc:(.text+0x212): undefined reference to `operator delete[](void*)'
/usr/bin/ld: stogo.cc:(.text+0x289): undefined reference to `operator delete[](void*)'
/usr/bin/ld: /home/linuxbrew/.linuxbrew/lib/libnlopt.a(stogo.cc.o): in function `Global::~Global()':
stogo.cc:(.text._ZN6GlobalD2Ev[_ZN6GlobalD5Ev]+0x40): undefined reference to `operator delete[](void*)'
/usr/bin/ld: stogo.cc:(.text._ZN6GlobalD2Ev[_ZN6GlobalD5Ev]+0x48): undefined reference to `operator delete(void*)'

```

File: Makefile [hunk 1]
```diff
@@ -110,6 +111,10 @@ server-static: build-web
 	rm -f $(BIN_OUTPUT_PATH)/viam-server
 	VIAM_STATIC_BUILD=1 GOFLAGS=$(GOFLAGS) go build $(LDFLAGS) -o $(BIN_OUTPUT_PATH)/viam-server web/cmd/server/main.go
 
+full-static:
```
Comment (Otterverse) [inline Makefile:114]: This needs to depend on `build-web` so the front end code gets included.

Comment (abe-winter) [reply Makefile]: done in ea5cebd63, sample run [here](https://github.com/viamrobotics/rdk/actions/runs/8883537628/job/24390599038)

**Review by Otterverse — APPROVED** [review]
LGTM

**Pull request merged**